metadata-ingestion/src/datahub/ingestion/source/kafka_connect/source_connectors.py

```diff
@@ -23,6 +24,339 @@
+        # Look for outbox tables in current topics
+        outbox_topics = [t for t in current_topics if "outbox" in t.lower()]
+
+        if not outbox_topics:
+            logger.warning(
+                "EventRouter transform applied but no outbox table found in source topics. "
+                "This may result in incomplete lineage."
+            )
+            return current_topics
+
+        # EventRouter produces outbox.event.{aggregatetype} format as documented
+        # This is a predictable transformation regardless of manifest topics
+        # since it's based on the event types extracted from the outbox table
+
+        # For the pipeline, we generate standard EventRouter output topics
+        # The actual event types will be determined by downstream RegexRouter or manifest topics
+
+        # Check if manifest contains EventRouter-style topics or further transformed topics
+        event_topics = []
+
+        # First, check for direct outbox.event.* pattern in manifest
+        direct_event_topics = [
+            t for t in manifest_topics if t.startswith("outbox.event.")
+        ]
+
+        if direct_event_topics:
+            event_topics = direct_event_topics
+        else:
+            # EventRouter with further transforms - this is complex and error-prone to guess
+            # The documented EventRouter behavior only guarantees outbox.event.* intermediate format
+            # Any further transforms are connector-specific and not predictable
+            logger.warning(
+                "EventRouter detected but no 'outbox.event.*' topics found in manifest. "
+                "This suggests EventRouter output is further transformed by other transforms. "
+                "For accurate lineage mapping with complex transform chains, "
+                "please use 'generic_connectors' config to specify explicit source-to-topic mappings."
+            )
+            # Return manifest topics as-is - don't attempt to guess transform patterns
+            event_topics = list(manifest_topics)
+
+        logger.info(
+            f"EventRouter mapping: {len(outbox_topics)} outbox tables -> {len(event_topics)} event topics"
+        )
+        return event_topics
+
+
+class RegexRouterTransform(BaseTransform):
+    """Kafka Connect RegexRouter transform."""
+
+    def apply(self, current_topics: List[str], manifest_topics: List[str]) -> List[str]:
+        """Apply RegexRouter transform to rename topics."""
+        regex_pattern = self.config.get("regex", "")
+        replacement = self.config.get("replacement", "")
+
+        if not regex_pattern or not replacement:
+            logger.warning("RegexRouter missing regex or replacement pattern")
+            return current_topics
+
+        transformed_topics = []
+
+        for topic in current_topics:
+            try:
+                # Use Java regex for consistency with existing Kafka Connect implementation
+                from java.util.regex import Pattern
```

Comment:
may this require to set new python dependencies for this source? in `metadata-ingestion/setup.py`

Reply:
JPype1 dependency is already defined: In setup.py line 496, the kafka-connect plugin includes "JPype1" as a dependency:
  "kafka-connect": sql_common | {"requests", "JPype1"},